src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -5852,7 +5187,14 @@ namespace VariableSpeedCoils {
                                            state.dataEnvrn->OutBaroPress,
                                            RoutineName);
 
-        MaxSpeed = state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).NumOfSpeeds;
+        auto &varSpeedCoil = state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum);
+
+        if (varSpeedCoil.availSched->getCurrentVal() <= 0.0) {
+            varSpeedCoil.SimFlag = false;
+            return;
+        }
```

Comment:
If the coil is scheduled off, set the `SimFlag` to `false` and return before doing the calc.